tests/e2e/vsphere_volume_expansion.go

```diff
@@ -1793,6 +1793,226 @@ var _ = ginkgo.Describe("Volume Expansion Test", func() {
 
 	})
 
+	/*
+		Verify online volume expansion  when CSI POD is down
+		1. Create StorageClass with allowVolumeExpansion set to true .
+		2. Create PVC which uses the StorageClass created in step 1.
+		3. Wait for PV to be provisioned.
+		4. Wait for PVC's status to become Bound and note down the size
+		5. Create a POD using the above created PVC
+		6. Bring down CSI controller POD
+		7. Modify PVC's size to trigger online volume expansion
+		8. verify the PVC status will change to "Resizing", but resize won't succeed untill CSI pod is up
+		9. Bringup the CSI controller POD
+		10. Verify PVC resize completes
+		8. Verify the resized PVC by doing CNS query
+		9. Make sure data is intact on the PV mounted on the pod
+		10.Make sure file system has increased
+	*/
+	ginkgo.It("[csi-supervisor] Verify online volume expansion when CSI POD is down", func() {
+		ginkgo.By("Invoking Test for Volume Expansion")
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		var vmUUID string
+		var pod *v1.Pod
+
+		ginkgo.By("Create StorageClass on shared VMFS datastore with allowVolumeExpansion set to true")
+		sharedVSANDatastoreURL := GetAndExpectStringEnvVar(envSharedDatastoreURL)
+		volHandle, pvclaim, pv, storageclass := createSCwithVolumeExpansionTrueAndDynamicPVC(f, client, sharedVSANDatastoreURL, storagePolicyName, namespace)
```

Comment:
Does envSharedDatastoreURL return VMFS datastore URL - I believe it returns VSAN datastore URL? - please correct